### DISCUSSION TITLE
Add / Modify HDR10 Metadata in the HEVC Bitstream

### DISCUSSION THREAD

**Discussion by ZimuQin** [quietvoid/dovi_tool#73] [Ideas]
Hi, I saw a new feature to discard HDR10+ metadata in the HEVC bitstream has been added. It's really helpful. 
I think maybe a feature to modify / add HDR10 metadata (especially the Mastering Display Color Volume information which would be used together with RPU) in HEVC SEI messages would be very useful when creating a compatible DV bitstream. Maybe it could be implemented to alter those data in BL while injecting the RPU?
Thank you for your great work!

**Comment by quietvoid** [comment]
It's too much extra complexity to add in this project.
It could be a new repo or an example in `hevc_parser`, but I'd probably go with a new repo.

**Reply by ZimuQin** [reply]
Does this mean you might implement it in the future? I'm looking forward to it, as I didn't find quite a neat way to do this except to re-encode BL. Thank you for all your great tools!

**Reply by quietvoid** [reply]
> Does this mean you might implement it in the future?

No, I have no use for this personally. And I don't have the time to work on it.

You could try this: https://github.com/SK-Hardwired/nv_hevc_hdr_patcher

**Reply by ZimuQin** [reply]
OK. Thank you anyway. I tried NVHSP before, but sadly I could not get it to work (it always shows some Python runtime error)...
